src/Silk.Core/Services/TagService.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Hosting;
+using Silk.Data.MediatR;
+using Silk.Data.Models;
+
+namespace Silk.Core.Services
+{
+    public class TagService 
+    {
+
+        private readonly IMediator _mediator;
+        public TagService(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        public async Task<Tag?> GetTagAsync(string tagName, ulong guildId)
+        {
+            Tag? dbTag = await _mediator.Send(new TagRequest.Get(tagName, guildId));
+
+            return dbTag;
+        }
+
+        public async Task<(bool success, string? reason)> AliasTagAsync(string tagName, string aliasName, ulong guildId, ulong ownerId)
```

Comment:
While I don't have anything against Tuples, I do think this could be done differently. How often is this function used?